<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CwmsGeneratorCodeEditorClass</class>
 <widget class="QDialog" name="CwmsGeneratorCodeEditorClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>569</width>
    <height>476</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Codegenerierung für Klasse</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Klassenname</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QLineEdit" name="m_pqleClassName"/>
   </item>
   <item row="2" column="0">
    <widget class="QCheckBox" name="m_pqchbGenerateStandardObjectList">
     <property name="text">
      <string>Standard Objektliste generieren</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Keyname Standard Objektliste</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1" colspan="2">
    <widget class="QLineEdit" name="m_pqleObjectListKeyname"/>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Zielverzeichnis</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLineEdit" name="m_pqleFilename"/>
   </item>
   <item row="4" column="2">
    <widget class="QPushButton" name="pushButton_3">
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="../resources/emresources.qrc">
       <normaloff>:/icons/search24.png</normaloff>:/icons/search24.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="3">
    <widget class="QGroupBox" name="m_pqgbWebInterface">
     <property name="title">
      <string>Web Interface generieren</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>3</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>3</number>
      </property>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Klassenname</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="m_pqleWebIfClassName"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QCheckBox" name="m_pqchbGenerateBLClass">
     <property name="text">
      <string>Business Logik Klasse generieren</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0" colspan="2">
    <widget class="QCheckBox" name="m_pqchbGenerateTemporaryDataClass">
     <property name="text">
      <string>Temporäre Datenklasse generieren</string>
     </property>
    </widget>
   </item>
   <item row="8" column="0" colspan="2">
    <widget class="QCheckBox" name="m_pqchbGenerateBaseClassMethods">
     <property name="text">
      <string>Funktionen für Basisklassen-Members generieren</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <widget class="QCheckBox" name="m_qpchbGenerateTestClass">
     <property name="text">
      <string>Test Klasse generieren</string>
     </property>
    </widget>
   </item>
   <item row="10" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Basisklasse verwenden</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Basisklassenname</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="m_pqleBaseClassName"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Dateiname der Basisklasse</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="m_pqleBaseClassFilename"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="11" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Programmiersprache</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="m_pqchbCplusplusQt">
        <property name="text">
         <string>C++ QT</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m_pqchbJavaGwt">
        <property name="text">
         <string>Java GWT</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="12" column="0" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/emresources.qrc">
         <normaloff>:/icons/cancel24.png</normaloff>:/icons/cancel24.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/emresources.qrc">
         <normaloff>:/icons/ok24.png</normaloff>:/icons/ok24.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>Package / Namespace</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="2">
    <widget class="QLineEdit" name="m_pqlePackage"/>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../resources/emresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>CwmsGeneratorCodeEditorClass</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>477</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>415</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>CwmsGeneratorCodeEditorClass</receiver>
   <slot>SearchClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>559</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>508</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>CwmsGeneratorCodeEditorClass</receiver>
   <slot>OKClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>532</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>473</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>SearchClickedSlot()</slot>
  <slot>OKClickedSlot()</slot>
 </slots>
</ui>
